network-api/networkapi/wagtailpages/tests/test_blog.py

```diff
@@ -50,11 +53,65 @@ def test_no_query(self):
         response = self.client.get(path=url)
 
         self.assertEqual(response.status_code, HTTPStatus.OK)
-        entries = response.context['entries'].specific()
+        entries = response.context["entries"].specific()
         self.assertIn(blog_page_7, entries)
         self.assertIn(blog_page_6, entries)
         self.assertIn(blog_page_5, entries)
         self.assertIn(blog_page_4, entries)
         self.assertIn(blog_page_3, entries)
         self.assertIn(blog_page_2, entries)
         self.assertNotIn(blog_page_1, entries)
+
+
+class TestBlogIndexAuthors(test_base.WagtailpagesTestCase):
+    def setUp(self):
+        super().setUp()
+        self.blog_index = blog_factories.BlogIndexPageFactory(parent=self.homepage)
+        self.blog_index_url = self.blog_index.get_url() + self.blog_index.reverse_subpage(
+            "blog_author_index"
+        )
+
+        self.profile_1 = profile_factories.ProfileFactory()
+        self.profile_2 = profile_factories.ProfileFactory()
+        self.profile_3 = profile_factories.ProfileFactory()
+
+        self.blog_page_1 = blog_factories.BlogPageFactory(
+            parent=self.blog_index, authors=[BlogAuthors(author=self.profile_1)]
+        )
+        self.blog_page_2 = blog_factories.BlogPageFactory(
+            parent=self.blog_index, authors=[BlogAuthors(author=self.profile_2)]
+        )
+
+    def test_route_success(self):
+        response = self.client.get(path=self.blog_index_url)
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+
+    def test_authors_rendered(self):
+        # Test that the blog index page renders blog authors
+        response = self.client.get(path=self.blog_index_url)
+        self.assertTemplateUsed(response, "wagtailpages/blog_author_index_page.html")
+        self.assertContains(response, self.profile_1.name)
+        self.assertContains(response, self.profile_2.name)
+        self.assertNotIn(self.profile_3.name, str(response.content))
```

Comment:
**Question** Would `self.assertNotContains` not work here?

Reply:
fixed in 022ab55e9ea317868ecb197b36199671add2f5f5 - thanks!